src/lib/providers/BidProvider.tsx

```diff
@@ -127,6 +146,40 @@ const BidProvider: React.FC<BidProviderType> = ({ children }) => {
 		}
 	};
 
+	//this will receive either DTOs, and will populate the parameters with the correct data
+	function getBidParameters(
+		DTO: NftIdBidsDto | AccountBidsDto,
+		missingString: string,
```

Comment:
Consider instead having two optional arguments to this function `account` and `tokenId`. You can default to what's in the dto, but if it's undefined, you use the optional argument.